File: python/example5_2.py
# Example 5.2
# Theil's Measure of Multicollinearity
%cd C:/Course19/ceR/python

import numpy as np
import pandas as pd
import statsmodels.formula.api as smf

data=pd.read_csv('http://web.pdx.edu/~crkl/ceR/data/longley.txt',sep='\s+',nrows=16)

model = smf.ols(formula='EM ~ YEAR + PGNP + GNP + AF',data = data).fit() 
print(model.summary())
R2 = model.rsquared 
R21 = smf.ols(formula='EM ~ PGNP + GNP + AF',data = data).fit().rsquared 
R22 = smf.ols(formula='EM ~ YEAR + GNP + AF',data = data).fit().rsquared 
R23 = smf.ols(formula='EM ~ YEAR + PGNP + AF',data = data).fit().rsquared 
R24 = smf.ols(formula='EM ~ YEAR + PGNP + GNP',data = data).fit().rsquared
theil = R2 - (R2 - R21) - (R2 - R22) - (R2 - R23) - (R2 - R24) 
print("Thiel's Measure of Multicollinearity = %f" % theil) 

File: R/example5_2.R
# Example 5.2
# Theil's Measure of Multicollinearity

setwd("C:/Course17/ceR/R")
data<-read.table("http://web.pdx.edu/~crkl/ceR/data/longley.txt",header=T,nrows=16)

year<-data$YEAR
pgnp<-data$PGNP
gnp<-data$GNP
af<-data$AF
em<-data$EM

results<-lm(em~year+pgnp+gnp+af)
results_summary<-summary(results)
results_summary
R2<-results_summary[8]

results<-lm(em~pgnp+gnp+af)
results_summary<-summary(results)
results_summary
R21<-results_summary[8]

results<-lm(em~year+gnp+af)
results_summary<-summary(results)
results_summary
R22<-results_summary[8]

results<-lm(em~year+pgnp+af)
results_summary<-summary(results)
results_summary
R23<-results_summary[8]

results<-lm(em~year+pgnp+gnp)
results_summary<-summary(results)
results_summary
R24<-results_summary[8]

print ("Theil's Measure of Multicollinearity =")

R2<-as(R2,"numeric")
R21<-as(R21,"numeric")
R22<-as(R22,"numeric")
R23<-as(R23,"numeric")
R24<-as(R24,"numeric")

R<-R2-((R2-R21)+(R2-R22)+(R2-R23)+(R2-R24))
R

rm(list=ls())
